src/main/java/org/broadinstitute/hellbender/cmdline/argumentcollections/VariantAnnotationArgumentCollection.java

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.hellbender.cmdline.argumentcollections;
+
+import org.broadinstitute.barclay.argparser.Advanced;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Arguments for requesting VariantContext annotations to be processed by {@link org.broadinstitute.hellbender.tools.walkers.annotator.VariantAnnotatorEngine}
+ * for tools that process variants objects.
+ */
+public class VariantAnnotationArgumentCollection implements Serializable {
```

Comment:
Please, consider to make this an abstract class to being able to hide some parameters from the command line. Then, a `DefaultGATKVariantAnnotationArgumentCollection` should be used in GATK with this implementation.

If the variant anotation engine is converted into a plugin, that will be useful for downstream projects: I was actually working on my own plugin, but I prefer to work with standards from the GATK framework, and if there are downstream projects that want to annotate variants with their own annotations, this will be required for example if they do not need groups to annotate together.

As an example, that will be similar to what I did for the `GATKReadFilterPluginDescriptor`  in https://github.com/broadinstitute/gatk/pull/2355. As an API user, I think that it is good to have all the arguments for the plugins as abstract classes to being able to hide some from the CLI...

Reply:
@magicDGS This would be a useful feature but it sounds like the benefit is mostly hidden behind pending work on converting annotations into a plugin, which we have planned for the 4.0 release. This change would probably be incorporated as part of that branch.

Reply:
I thought that this could be addressed here for using the `VariantAnnotationArgumentCollection` with the `VariantAnnotator` too, even if the plugin is not there. I cannot hide some parameters with the current implementation, to being able to pass the argument collection to the annotator...

Anyway, I can submit a patch if there is no planned timeplan for the plugin, to being able to use something similar in my code...

Reply:
Maybe it would be easier to just add a generic/universal way to hide arguments in barclay, instead of having to complicate all of our argument collections with added abstract classes?

Reply:
I think that the main reason of having argument collections is for tweak the arguments for 
make optional/required some arguments (such as `ReferenceInputArgumentCollection`) or have the ability to tweak some arguments to be hidden/removed from the user.

I believe that with the current system for parsing arguments and argument collections is going to be complicated to hide arguments. In addition, sometimes the requirement isn't hide the argument but provide a tool-specific documentation (e.g., the `ReadFilter` plugin specifies that are filters __before__ the analysis; in some cases, I have tools which perform the filtering __after__ processing, and it is important for the user to know that difference).

I think that having interfaces for a framework is the best way to encourage users with specific needs to use it, unless you really want to forbid the usage/tweak of some components. I see several use cases for having the abstract argument collections....